src/tiled/eyevisibilitydelegate.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * eyevisibilitydelegate.cpp
+ * Copyright 2008-2017, Thorbjørn Lindeijer <thorbjorn@lindeijer.nl>
+ * Copyright 2010, Andrew G. Crowell <overkill9999@gmail.com>
+ * Copyright 2010, Jeff Bland <jksb@member.fsf.org>
+ * Copyright 2011, Stefan Beller <stefanbeller@googlemail.com>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "layerdock.h"
+#include "eyevisibilitydelegate.h"
+#include "layer.h"
+#include "layermodel.h"
+#include "map.h"
+#include "mapdocument.h"
+#include "mapdocumentactionhandler.h"
+#include "objectgroup.h"
+#include "reversingproxymodel.h"
+#include "utils.h"
+
+#include <QBoxLayout>
+#include <QApplication>
+#include <QContextMenuEvent>
+#include <QLabel>
+#include <QMenu>
+#include <QPainter>
+#include <QSlider>
+#include <QUndoStack>
+#include <QToolBar>
+
+using namespace Tiled;
+using namespace Tiled::Internal;
+
+EyeVisibilityDelegate::EyeVisibilityDelegate(QObject *parent):
+    QItemDelegate(parent),
+    mVisiblePixmap(QLatin1String(":/images/12x12/layer-visible.png")),
+    mInvisiblePixmap(QLatin1String(":/images/12x12/layer-invisible.png"))
+{
+}
+
+void EyeVisibilityDelegate::drawCheck(QPainter *painter, const QStyleOptionViewItem &option,
+    const QRect &rect, Qt::CheckState state) const
+{
+    Q_UNUSED(option)
+    if (state == Qt::Checked)
+    {
```

Comment:
Coding style: Please don't use `{` and `}` for single-line bodies (also for multi-line bodies, they should not be on their own line, except for functions).